x/gov/keeper/msg_server.go

```diff
@@ -312,3 +405,36 @@ func (k legacyMsgServer) Deposit(goCtx context.Context, msg *v1beta1.MsgDeposit)
 	}
 	return &v1beta1.MsgDepositResponse{}, nil
 }
+
+func validateAmount(amount sdk.Coins) error {
+	if !amount.IsValid() {
+		return sdkerrors.ErrInvalidCoins.Wrap(amount.String())
+	}
+
+	if !amount.IsAllPositive() {
+		return sdkerrors.ErrInvalidCoins.Wrap(amount.String())
+	}
+
+	return nil
+}
+
+func validateMsgs(msg *v1.MsgSubmitProposal) error {
+	msgs, err := msg.GetMsgs()
```

Comment:
Usually, in the message server we already call `GetMsgs()`, we can remove it here and have this function takes a `[]sdk.Msg`. This avoid getting messages twice.